Lib/tarfile.py

```diff
@@ -895,6 +895,9 @@ class TarInfo(object):
         _link_target = None,
         )
 
+    _name_uid0 = None    # Cached uname of uid=0
+    _name_gid0 = None    # Cached gname of gid=0
```

Comment:
What I meant before is: why using a class variable? the issue is that once we deduce `uid=0`, we're stuck with it for the *entire* Python process. Instead, these should be instance attributes and not class attributes. Similar to the current `self._unames` variables.